src/Entity/Item.php

```diff
@@ -83,21 +83,18 @@ public function getId() {
 	}
 
 	/**
-	 * Can be integer since 0.1.
-	 * Can be ItemId since 0.5.
-	 * Can be null since 1.0.
+	 * @since 0.5, can be null since 1.0
 	 *
 	 * @param ItemId|null $id
 	 *
 	 * @throws InvalidArgumentException
 	 */
 	public function setId( $id ) {
-		if ( $id === null || $id instanceof ItemId ) {
-			$this->id = $id;
-		} else {
-			throw new InvalidArgumentException( '$id must be an instance of ItemId, an integer,'
-				. ' or null' );
+		if ( !( $id instanceof ItemId ) && $id !== null ) {
```

Comment:
Ok. But then the assignment would be in the if, and the exceptional case in the else (or below the if). I wanted to use the guard pattern here, with the exceptional case first.